src/Planner/PlannerJoins.cpp

```diff
@@ -528,7 +528,7 @@ JoinClausesAndActions buildJoinClausesAndActions(
         size_t join_clause_key_nodes_size = join_clause.getLeftKeyNodes().size();
 
         if (join_clause_key_nodes_size == 0)
-            throw Exception(ErrorCodes::INVALID_JOIN_ON_EXPRESSION, "JOIN {} cannot get JOIN keys",
+            throw Exception(ErrorCodes::INVALID_JOIN_ON_EXPRESSION, "Cannot determine join keys in {}",
```

Comment:
Unrelated to the changes, but there is an extra space for some reason:
```
Code: 403. DB::Exception: Cannot determine join keys in  INNER JOIN ... ON (t1.x = t2.x) OR (t1.x IS NULL). (INVALID_JOIN_ON_EXPRESSION)
```

Can't find why. We can fix it in the future